src/hooks/user/usePermission.ts

```diff
@@ -0,0 +1,95 @@
+import { useMemo } from "react";
+
+import { ALWAYS_EMPTY_ARRAY, COLLECTION_ROLES } from "settings";
+
+import { SpaceId, UserId, WorldId } from "types/id";
+
+import { useLiveDocument } from "hooks/fire/useLiveDocument";
+import { useSpacesByOwner } from "hooks/spaces/useSpacesByOwner";
+import { useWorldsByOwner } from "hooks/worlds/useWorldsByOwner";
+
+type AdminRole = {
+  allowAll: boolean;
+  users: string[];
+};
+
+type UsePermissionOptions = {
+  userId: UserId;
+  worldId?: WorldId;
+  spaceId?: SpaceId;
+};
+
+export const usePermission = (options: UsePermissionOptions) => {
+  const path = useMemo(() => [COLLECTION_ROLES, "admin"], []);
```

Comment:
Could this just be a boring constant outside of the component?

Reply:
I think I have figured out a way and place for these.

Reply:
945342eb8af0a1591a833f0c785c70b1c61b60aa `PATH.rolesAdmin` from `apiSettings.ts` More paths can simply be added to `PATH` and will be constant array values.